spark-cassandra-connector-java/src/main/java/com/datastax/spark/connector/japi/RDDJavaFunctions.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    /**
+     * Applies a function to each item, and groups consecutive items having the same value together.
+     * Contrary to {@code groupBy}, items from the same group must be already next to each other in the
+     * original collection. Works locally on each partition, so items from different partitions will
+     * never be placed in the same group.
+     */
+    public <U> JavaPairRDD<U, Iterable<T>> spanBy(final Function<T, U> f, ClassTag<U> keyClassTag) {
+        ClassTag<Tuple2<U, Iterable<T>>> tupleClassTag = classTag(Tuple2.class);
+        ClassTag<Iterable<T>> iterableClassTag = CassandraJavaUtil.classTag(Iterable.class);
+
+        RDD<Tuple2<U, Iterable<T>>> newRDD = rddFunctions.spanBy(toScalaFunction1(f))
+                .map(JavaApiHelper.<U, T, scala.collection.Iterable<T>>valuesAsJavaIterable(), tupleClassTag);
+
+        return new JavaPairRDD<>(newRDD, keyClassTag, iterableClassTag);
+    }
+
+    /**
+     * Uses the data from {@code RDD} to join with a Cassandra table without retrieving the entire table.
+     * Any RDD which can be used to saveToCassandra can be used to joinWithCassandra as well as any RDD
+     * which only specifies the partition Key of a Cassandra Table. This method executes single partition
+     * requests against the Cassandra Table and accepts the functional modifiers that a normal
+     * {@link CassandraTableScanRDD} takes.
+     * <p/>
+     * By default this method only uses the Partition Key for joining but any combination of columns
+     * which are acceptable to C* can be used in the join. Specify columns using joinColumns as a
+     * parameter or the {@code on()} method.
+     */
+    public <R> CassandraJavaPairRDD<T, R> joinWithCassandraTable(
+            String keyspaceName,
+            String tableName,
+            ColumnSelector selectedColumns,
+            ColumnSelector joinColumns,
+            RowReaderFactory<R> rowReaderFactory,
+            RowWriterFactory<T> rowWriterFactory
+    ) {
+        ClassTag<T> classTagT = rdd.toJavaRDD().classTag();
+        ClassTag<R> classTagR = JavaApiHelper.getClassTag(rowReaderFactory.targetClass());
+
+        CassandraConnector connector = defaultConnector();
+        Option<ClusteringOrder> clusteringOrder = Option.empty();
+        Option<Object> limit = Option.empty();
+        CqlWhereClause whereClause = CqlWhereClause.empty();
+        ReadConf readConf = ReadConf.fromSparkConf(rdd.conf());
+
+        CassandraJoinRDD<T, R> joinRDD = new CassandraJoinRDD<>(
+                rdd,
+                keyspaceName,
+                tableName,
+                connector,
+                selectedColumns,
+                joinColumns,
+                whereClause,
+                limit,
+                clusteringOrder,
+                readConf,
+                classTagT,
+                classTagR,
+                rowWriterFactory,
+                rowReaderFactory);
```

Comment:
This is something I'd like to fix in the next round of code cleanup. Those parameter lists are getting crazy.